ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthFeeHistoryTest.java

```diff
@@ -179,14 +187,51 @@ public void shouldBoundRewardsCorrectly() {
             .upperBoundGasAndPriorityFeeCoefficient(500L)
             .build(); // Max reward = Wei.One * 500L / 100 = 5.0
 
+    final var blockchainQueries = mockBlockchainQueries(blockchain, Wei.of(7));
+    when(miningCoordinator.getMinPriorityFeePerGas()).thenReturn(Wei.ONE);
+
     EthFeeHistory ethFeeHistory =
-        new EthFeeHistory(null, blockchain, miningCoordinator, apiConfiguration);
+        new EthFeeHistory(null, blockchainQueries, miningCoordinator, apiConfiguration);
 
-    List<Wei> rewards = ethFeeHistory.computeRewards(rewardPercentiles, block);
+    List<Wei> rewards = ethFeeHistory.computeRewards(rewardPercentiles, block, Wei.of(7));
 
     // Define the expected bounded rewards for each percentile
     List<Wei> expectedBoundedRewards = Stream.of(2, 2, 2, 4, 5, 5, 5, 5, 5).map(Wei::of).toList();
-    assertThat(expectedBoundedRewards).isEqualTo(rewards);
+    assertThat(rewards).isEqualTo(expectedBoundedRewards);
+  }
+
+  @Test
+  public void shouldApplyLowerBoundRewardsCorrectly() {
+    // This test checks that the rewards are correctly bounded by the lower and upper limits,
+    // when the calculated lower bound for the priority fee is greater than the configured one.
+    // Configured minPriorityFeePerGas is 0 wei, minGasPrice is 10 wei and baseFee is 8 wei,
+    // so for a tx to be mined the minPriorityFeePerGas is raised to 2 wei before applying the
+    // coefficients.
+
+    List<Double> rewardPercentiles =
+        Arrays.asList(0.0, 5.0, 10.0, 27.50, 31.0, 59.0, 60.0, 61.0, 100.0);
+
+    Block block = mock(Block.class);
+    Blockchain blockchain = mockBlockchainTransactionsWithPriorityFee(block);
+
+    ApiConfiguration apiConfiguration =
+        ImmutableApiConfiguration.builder()
+            .isGasAndPriorityFeeLimitingEnabled(true)
+            .lowerBoundGasAndPriorityFeeCoefficient(200L) // Min reward = 2 * 200L / 100 = 4.0
+            .upperBoundGasAndPriorityFeeCoefficient(300L)
+            .build(); // Max reward = 2 * 300L / 100 = 6.0
+
+    final var blockchainQueries = mockBlockchainQueries(blockchain, Wei.of(10));
+    when(miningCoordinator.getMinPriorityFeePerGas()).thenReturn(Wei.ZERO);
+
+    EthFeeHistory ethFeeHistory =
+        new EthFeeHistory(null, blockchainQueries, miningCoordinator, apiConfiguration);
+
+    List<Wei> rewards = ethFeeHistory.computeRewards(rewardPercentiles, block, Wei.of(8));
+
+    // Define the expected bounded rewards for each percentile
+    List<Wei> expectedBoundedRewards = Stream.of(4, 4, 4, 4, 5, 6, 6, 6, 6).map(Wei::of).toList();
```

Comment:
I don't get this test. If lower bound is set to 10 Wei, why are we getting values lower than that?

Reply:
in this case `minGasPrice` is 10wei, and since `baseFee` is 8wei, the calculated min priority fee (necessary to get the tx minable) is 2wei that is greater than the configured `minPriorityFeePerGas=0` so the former is used.

Then the rewards (effective priority fees) are bounded by 200 and 300 coefficients, so the rewards must be in the interval 4wei and 6wei

Reply:
Ok, I'm following it now, thanks!